### ISSUE TITLE
config option for unused namespaces without as / refer

### ISSUE THREAD

**Issue by jakemcc** [clj-kondo/clj-kondo#1680]
**version**

```
$ clj-kondo --version
clj-kondo v2022.04.25
```

**platform**

MacOS. Using a prebuilt native version.

**problem**

Unused `:require`s in an `ns` form are not detected unless they use `:refer`

**repro**

```
$ cat repro.clj
(ns repro
  (:require [clojure.data]))
$ clj-kondo --lint repro.clj
linting took 14ms, errors: 0, warnings: 0

$ cat repro.clj
(ns repro
  (:require clojure.data))
$ clj-kondo --lint repro.clj
linting took 15ms, errors: 0, warnings: 0
```

Working with a `:refer`

```
$ cat repro.clj
(ns repro
  (:require [clojure.data :refer [diff]]))
 
$ clj-kondo --lint repro.clj
repro.clj:2:14: warning: namespace clojure.data is required but never used
repro.clj:2:35: warning: #'clojure.data/diff is referred but never used
linting took 14ms, errors: 0, warnings: 2
```

**config**

I did the example without any configuration.

**expected behavior**

I'd expect it to catch that the required namespace is unused.

**Comment by borkdude** [comment]
I improved the docs here: https://github.com/clj-kondo/clj-kondo/blob/master/doc/linters.md#unused-namespace

**Comment by jakemcc** [comment]
Thanks for the details.

I guess I would have expected folks to take advantage of the `:exclude` configuration option for listing namespaces they may be loading for side effects. Seems like you'd need to do that anyway, in case you start requiring in side-effecting namespaces w/ `:as` or `:refer`.

I generally dislike suggesting a project take on _more_ configuration options, since I know what a pain that can be for a maintainer, but would another option to `:unused-namespaces` turning on identifying unused non-referred `:requires` have potential to be merged? If so, I'm happy to figure out how to enable that if I could get a pointer to where you think the change might need to be made.

As some background, the team I recently joined prefers to use the full namespace (so `clojure.set/union` instead of `set/union`) so the project is full of `:require` without `:as` or `:refer`. This is currently the one linter that is keeping us from switching to clj-kondo (~3 seconds to run on our largest service's codebase) from eastwood (~60 seconds on the same codebase).

**Comment by borkdude** [comment]
@jakemcc OK, let's make that a config option.

**Comment by jakemcc** [comment]
@borkdude I looked around some more to see what it might take to implement and ran into around some and I think this might actually already be supported with the `:simple-libspec` option.

```
$ cat repro.clj
(ns repro
  (:require [clojure.data]
            clojure.xml
            [clojure.string :as string]))

$ clj-kondo --lint repro.clj --config '^:replace {:linters {:unused-namespace {:simple-libspec false :level warn}}}'
repro.clj:4:14: warn: namespace clojure.string is required but never used

$ clj-kondo --lint repro.clj --config '^:replace {:linters {:unused-namespace {:simple-libspec true :level warn}}}'
repro.clj:2:14: warn: namespace clojure.data is required but never used
repro.clj:3:13: warn: namespace clojure.xml is required but never used
repro.clj:4:14: warn: namespace clojure.string is required but never used
```

Changing the file to use `clojure.data`

```
$ cat repro.clj
(ns repro
  (:require [clojure.data]
            clojure.xml
            [clojure.string :as string]))

(clojure.data/diff [1 2 3] [2 3])

$ clj-kondo --lint repro.clj --config '^:replace {:linters {:unused-namespace {:simple-libspec true :level warn}}}'
repro.clj:3:13: warn: namespace clojure.xml is required but never used
repro.clj:4:14: warn: namespace clojure.string is required but never used
```

**Comment by borkdude** [comment]
@jakemcc You're right, I totally forgot about that option!

**Comment by jakemcc** [comment]
This turned into the best kind of accepted feature request, the one that
already exists 🙂

**Comment by borkdude** [comment]
Thanks for the doc updates! Closing.
